### DISCUSSION TITLE
How to render images enabling DLSS in a headless device?

### DISCUSSION THREAD

**Discussion by aritra-pal** [NVlabs/instant-ngp#657] [Q&A]
Hi, first of all, thanks for this awesome work. I have built it successfully on my device with vulkan SDK. So, I can enable DLSS in the GUI while rendering the scenes. 
However, I am wondering while I am saving images using the run.py script are those images rendered with DLSS?
If not, can you please help me to enable DLSS while using the command line?

**Comment by Tom94** [comment]
Hi there, DLSS is not supported in headless mode. Supporting it would require disentangling Vulkan from GLFW, as well as figuring out an API to carry over motion vectors across separate `testbed.render` calls. Both of which are out of scope of this project at the moment. Cheers!

**Reply by aritra-pal** [reply]
Thanks for your quick response. In that case, I would like to save DLSS rendered images from the GUI.  Can you please suggest me how can I do that ?

**Reply by Tom94** [reply]
While the GUI doesn't have a button to save screenshots, you can use the interactive Python REPL for that.
1. Use `scripts/run.py` to launch the GUI and train. Then move the camera to the position you'd like to screenshot.
2. Press the GUI button for entering the Python REPL
3. Execute
```py
testbed.display_gui = False
image = testbed.screenshot()
write_image("bla.png", image)
```
